lib/metric-config-parser/metric_config_parser/tests/test_config.py

```diff
@@ -505,6 +507,92 @@ def test_invalid_metric_level(self):
                 last_modified=datetime.datetime.now(),
             )
 
+    def test_valid_aggregation_units(self):
+        config_str = dedent(
+            """
+            [metrics.active_hours]
+            select_expression = "1"
+            data_source = "baseline"
+            owner = "me@example.com"
+            category = "test"
+            aggregation_units = ["profile_group_id"]
+
+            [metrics.other_metric]
+            select_expression = "1"
+            data_source = "baseline"
+            owner = ["me@example.com", "you@example.com"]
+            category = "test"
+            aggregation_units = ["client_id", "profile_group_id"]
+
+            [data_sources.baseline]
+            from_expression = "mozdata.search.baseline"
+            experiments_column_type = "simple"
+            aggregation_units = ["client_id", "profile_group_id"]
+            """
+        )
+
+        definition = DefinitionConfig(
+            slug="firefox_desktop",
+            platform="firefox_desktop",
+            spec=AnalysisSpec.from_dict(toml.loads(config_str)),
+            last_modified=datetime.datetime.now(),
+        )
+        config_collection = ConfigCollection(
+            configs=[], outcomes=[], defaults=[], definitions=[definition]
+        )
+
+        metric_definition = config_collection.get_metric_definition(
+            "active_hours", "firefox_desktop"
+        )
+        assert metric_definition.aggregation_units == [AnalysisUnit.PROFILE_GROUP]
+
+        metric_definition = config_collection.get_metric_definition(
+            "other_metric", "firefox_desktop"
+        )
+        assert metric_definition.aggregation_units == [
+            AnalysisUnit.CLIENT,
+            AnalysisUnit.PROFILE_GROUP,
+        ]
+
+        data_source_definition = config_collection.get_data_source_definition(
+            "baseline", "firefox_desktop"
+        )
+        assert data_source_definition.aggregation_units == [
+            AnalysisUnit.CLIENT,
+            AnalysisUnit.PROFILE_GROUP,
+        ]
+
+    @pytest.mark.parametrize(
+        "metric_units,ds_units",
+        (
+            ("aggregation_units = 'client_id'", ""),
+            ("aggregation_units = ['invalid']", ""),
+            ("", "aggregation_units = 'client_id'"),
+            ("", "aggregation_units = ['invalid']"),
+        ),
+    )
+    def test_invalid_aggregation_units(self, metric_units, ds_units):
+        config_str = dedent(
+            f"""
+            [metrics.active_hours]
+            select_expression = "1"
+            data_source = "baseline"
+            {metric_units}
+
+            [data_sources.baseline]
+            from_expression = "mozdata.search.baseline"
+            experiments_column_type = "simple"
+            {ds_units}
+            """
+        )
+        with pytest.raises(Exception):
```

Comment:
Is there any way to make this error more specific? `Exception` seems vague

Reply:
Yea I didn't like this either. The exception is a `ValueError`, but it's actually multiple `ValueError`s within a group of exceptions. For example, because it expects a list, it parses a string value as a list of characters, none of which are valid `AnalysisUnit`s, and each of those characters produces a `ValueError`, raised together in a group. In Python 3.11 there was an `ExceptionGroup` introduced, which might give us more options, but I opted for the overly broad solution here in line with other parts of the code.;